cmd/config.go

```diff
@@ -309,6 +301,82 @@ type CAADistributedResolverConfig struct {
 	Proxies     []string
 }
 
+// TemporalLogDescription describes a single shard of a temporally shared
+// CT log
+type TemporalLogDescription struct {
+	URI         string
+	Key         string
+	WindowStart time.Time
+	WindowEnd   time.Time
+}
+
+// TemporalSet contains a set of temporal shards of a single log
+type TemporalSet struct {
+	Name   string
+	Shards []TemporalLogDescription
+}
+
+// Setup initializes the TemporalSet by parsing the start and end dates
+// and verifying WindowEnd > WindowStart
+func (ts *TemporalSet) Setup() error {
+	if ts.Name == "" {
+		return errors.New("Name cannot be empty")
+	}
+	if len(ts.Shards) == 0 {
+		return errors.New("temporal set contains no shards")
+	}
+	for i := range ts.Shards {
+		if ts.Shards[i].WindowEnd.Before(ts.Shards[i].WindowStart) || ts.Shards[i].WindowEnd.Equal(ts.Shards[i].WindowStart) {
```

Comment:
Please wrap this line. Then it will be easier to see the similarity between the clauses.